### PR TITLE
Use accurate meter size in Web Mercator projection

### PR DESCRIPTION
**Pull request by Pessimistress**
For #6181 & discussion https://github.com/visgl/deck.gl/discussions/6101

Before:

deck gl-project_size-old

After:

deck gl-project_size-new

I think the perf impact is acceptable (approximately 1 additional `cos` instruction per vertex at zoom < 12); the main question is how big of a breaking change this is. Is it major-release-worthy? All existing render tests pass as-is; the applications that will break visually are the ones using meter sizes on a global scale. @heshan0131 what is the default radius units in Kepler.gl?

#### Change List
- `project` shader module `project_size` returns result based on vertex position
- `WebMercatorViewport.projectPosition` returns result based on latitude
- Separate shader code paths for pixel size and meter size -- we used to differentiate `*Units: 'pixels'` and `*Units: 'meters'` by adjusting the `*Scale` uniform. This no longer works if meter size is non-linear.
- Render test

#### TODO

- [ ] Documentation / Upgrade guide

### CODE REVIEW

**Comment by Pessimistress** [comment]
Since this is all done in the project module, I think it's possible to make this an opt-in change by introducing a new uniform.

File: modules/core/src/shaderlib/project/project.glsl.js [hunk 1]
```diff
@@ -55,20 +55,47 @@ const vec3 ZERO_64_LOW = vec3(0.0);
...
+float project_size() {
+  if (project_uProjectionMode == PROJECTION_MODE_WEB_MERCATOR &&
+    project_uCoordinateSystem == COORDINATE_SYSTEM_LNGLAT) {
+
+    // uCommonUnitsPerMeter in low-zoom Web Mercator is non-linear
+    // Adjust by 1 / cos(latitude)
+    // If geometry.position (vertex in common space) is populated, use it
+    // Otherwise use geometry.worldPosition (anchor in world space)
+    
+    if (geometry.position.w == 0.0) {
```
Comment (ibgreen) [inline modules/core/src/shaderlib/project/project.glsl.js:68]: Can this be precalculated so we don't need if statements (they tend to slow things down)?

File: modules/layers/src/text-layer/text-background-layer/text-background-layer-vertex.glsl.js [hunk 2]
```diff
@@ -46,7 +47,7 @@ void main(void) {
 
   // project meters to pixels and clamp to limits
   float sizePixels = clamp(
-    project_size_to_pixel(instanceSizes * sizeScale),
+    pixelSize ? instanceSizes * sizeScale : project_size_to_pixel(instanceSizes * sizeScale),
```
Comment (ibgreen) [inline modules/layers/src/text-layer/text-background-layer/text-background-layer-vertex.glsl.js:50]: Would it make sense to add a pixelSize argument to the `project function` and remove the if statement in all the shaders?

File: test/modules/core/shaderlib/project/project-glsl.spec.js [hunk 3]
```diff
@@ -65,58 +65,46 @@ const OUT_BUFFER = new Buffer(gl, 16);
 
 // used in printing a float into GLSL code, 1 will be 1.0 to avoid GLSL compile errors
 const MAX_FRACTION_DIGITS = 5;
+function getScalerType(a) {
```
Comment (ibgreen) [inline test/modules/core/shaderlib/project/project-glsl.spec.js:68]: Maybe slightly more descriptive name for this function, or some JSDoc?

**Review by ibgreen — APPROVED** [review]

**Comment by ibgreen** [comment]
As always it would be awesome to have a perf bench to measure actual performance impact on shaders. It looks fine, but we keep adding features and there is always a nagging concern.

File: modules/core/src/shaderlib/project/project.glsl.js (hunk 1 above)
Comment (Pessimistress) [reply modules/core/src/shaderlib/project/project.glsl.js:68]: Not without breaking existing layers. 

This test will yield the same result for all vertices in a given layer, so there won't be any sync points.

**Pull request merged**